### ISSUE TITLE
Right-clicking to delete a save crashes the game about 75% of the time

### ISSUE THREAD

**Issue by 3shirts** [OpenLoco/OpenLoco#650]
OS: Windows 10
Version: 20.07-93
Commit/Build:  7c46804

When in the load or save window if I right click to delete a save, the game will crash more often than not. I would estimate 75% of the time. It does delete the save successfully however.

Steps to reproduce:
1. Create a save game.
2. Go into the load or save window and right click the save.
3. Click yes to confirm deletion.

Is the issue reproducible in Locomotion (vanilla)?
No

**Comment by AaronVanGeffen** [comment]
Thanks for the report. The load/save window has only been partially implemented in C++; we're still relying on the original game's routines for several routines, among which the mouse down and over functions relating to the file list:
```cpp
        _events.scroll_mouse_down = reinterpret_cast<void (*)(window*, int16_t, int16_t, uint8_t)>(0x004464F7);
        _events.scroll_mouse_over = reinterpret_cast<void (*)(window*, int16_t, int16_t, uint8_t)>(0x004464B1);
```
This should be solved by implementing these routines. Admittedly, I am a little surprised to learn you could delete save files just by right-clicking them, though!

**Comment by AaronVanGeffen** [comment]
I have implemented both event functions in #666. Could you download a CI build and test whether this resolves the issue for you?

**Comment by AaronVanGeffen** [comment]
I'm closing this on the assumption this no longer happens in v20.10.
